dodola/tests/test_services.py

```diff
@@ -443,12 +448,16 @@ def test_regrid_weights_integration(domain_file, tmpdir):
     assert actual_shape == expected_shape
 
 
-def test_clean_cmip6():
-    """Tests that cmip6 cleanup removes extra dimensions on dataset"""
+@pytest.mark.parametrize(
+    "gcm_variable", [pytest.param("tasmax"), pytest.param("tasmin"), pytest.param("pr")]
+)
+def test_clean_cmip6(gcm_variable):
+    """Tests that cmip6 cleanup removes extra dimensions on dataset
+    and that precip units are converted if variable is precip"""
     # Setup input data
     n = 1500  # need over four years of daily data
     ts = np.sin(np.linspace(-10 * np.pi, 10 * np.pi, n)) * 0.5
-    ds_gcm = _gcmfactory(ts, start_time="1950-01-01")
+    ds_gcm = _gcmfactory(ts, gcm_variable=gcm_variable, start_time="1950-01-01")
```

Comment:
I'd add the logic for adding the "units" attrib for "pr" in the input data just under here in the new version of the test.

Reply:
Done.